compression.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+const (
+	CompressionNone int8 = iota
+	CompressionGZIP
+	CompressionSnappy
+	CompressionLZ4
```

Comment:
I had suggested you to move those constants in the packages that implement the compression codecs with a value like this:
```go
package gzip
// Having the code called the same in every package makes the code very programmer friendly
// as it becomes intuitive what it's gonna be named in each package. It also gives a clear
// guideline to future maintainers about how they should structure packages that provide
// compression codec implementations.
const Code = 1
```
Any reason why you'd prefer to keep those constants here?

Reply:
If the constant is in the snappy package it would be snappy.Code